haskell/test/Test/SegCFMM.hs

```diff
@@ -1,16 +1,106 @@
 -- SPDX-FileCopyrightText: 2021 Arthur Breitman
 -- SPDX-License-Identifier: LicenseRef-MIT-Arthur-Breitman
 
+{-# LANGUAGE NumericUnderscores #-}
+{-# OPTIONS_GHC -Wno-type-defaults #-}
+
 module Test.SegCFMM
   ( test_SegCFMM
   ) where
 
 import Universum
 
+import Lorentz hiding (assert, now, (>>))
+import Michelson.Typed.Convert (convertContract, untypeValue)
+import Morley.Nettest
+import Morley.Nettest.Tasty
 import Test.Tasty (TestTree, testGroup)
+import Tezos.Address (unsafeParseAddress)
+import Tezos.Core
+
+import SegCFMM.Types
+import Test.SegCFMM.Contract (segCFMMContract)
+import Test.SegCFMM.Storage (defaultStorage)
 
 {-# ANN module ("HLint: ignore Reduce duplication" :: Text) #-}
 
+
 test_SegCFMM :: TestTree
 test_SegCFMM = testGroup "Segmented CFMM Tests"
-  []
+  [ nettestScenarioCaps "Set a position"
+      setPositionTest
+  , nettestScenarioCaps "Swap X for Y"
+      swapXYTest
+  ]
+
+setPositionTest
+  :: MonadNettest caps base m
+  => m ()
+setPositionTest = do
+  owner1 <- newAddress auto
+  cfmm <- originateSegCFMM defaultStorage
+  let
+    -- TODO: originate proper contract
+    addr = unsafeParseAddress "KT1MPGAKt68xEgLe1c2n7AG89Hph8Q7o44UX"
+
+    param = SetPositionParam
+          { sppLowerTickIndex = TickIndex 2
+          , sppUpperTickIndex = TickIndex 10
+          , sppLowerTickWitness = TickIndex $ negate infinityInt
+          , sppUpperTickWitness = TickIndex $ negate infinityInt -- otherwise fail with 100 error code
+          , sppLiquidityDelta = 0
+          , sppToX = addr
+          , sppToY = addr
+          }
+
+  withSender owner1 $
+    call cfmm (Call @"Set_position") param
+  -- ^ TODO: getting 300 error (tick not initialized)
+
+  _ <- getStorage @Storage (unTAddress cfmm)
+  pure ()
+
+swapXYTest
+  :: MonadNettest caps base m
+  => m ()
+swapXYTest = do
+  owner1 <- newAddress auto
+  receiver <- newAddress auto
+  cfmm <- originateSegCFMM defaultStorage
+
+  let
+    param = XToYParam
+          { xpDx = pickX (mkX 10 :: X 80 Natural)
+          , xpDeadline = parseTimestamp "2022-01-01T00:00:00Z" & fromMaybe (error "shouldn't happen.")
```

Comment:
You also can use `timestampQuote` quasi-quoter.

Reply:
Nice, updated this now.